drivers/syslog/README.txt

```diff
@@ -120,21 +120,31 @@ SYSLOG Channels
   SYSLOG channel is represented by an interface defined in
   include/nuttx/syslog/syslog.h:
 
-    /* This structure provides the interface to a SYSLOG device */
+    /* SYSLOG I/O redirection methods */
 
-    typedef CODE int (*syslog_putc_t)(int ch);
-    typedef CODE int (*syslog_flush_t)(void);
+    typedef CODE ssize_t (*syslog_write_t)(FAR struct syslog_channel_s *channel,
+                                       FAR const char *buf, size_t buflen);
+    typedef CODE int (*syslog_putc_t)(FAR struct syslog_channel_s *channel,
+                                  int ch);
+    typedef CODE int (*syslog_flush_t)(FAR struct syslog_channel_s *channel);
+
+    /* SYSLOG device operations */
+
+    struct syslog_channel_ops_s
+    {
+      syslog_putc_t  sc_putc;   /* Normal buffered output */
```

Comment:
@fjpanag we can create a new patch to remove sc_putc if all driver implement sc_write.

Reply:
@xiaoxiang781216 I had a look at this, but it is not as simple as it seems...

`nx_vsyslog()` uses `lib_vsprintf()` which only uses a putc function. And thus this is the only function used, apart from some special cases (like flushing the buffer), that use `sc_write()`.

Unless you have a good idea on how this can be done, I would prefer to leave it as is.  
And possibly work on this, when whole issue of syslog thread-safety is resolved, some time in the near future.

Reply:
We can call sc_write with one char.

Reply:
But what would be the advantage of this?

My initial thought of removing `sc_put()` was under the condition that the logger will be able to deliver the whole log line to `sc_write()` so it can be written atomically. Calling `sc_write()` with a single character does not introduce any advantage.

Reply:
Yes, it's better to remove sc_put and ensure we call sc_write with the whole line.

Reply:
I don't fully understand the purpose of `put` either, but my guess is that is done this way so that it is possible to print the message from interrupt handlers without having to hold any lock (there is some comment about this in syslog implementation). When I looked at how to make ramlog implement write() I got stuck with this.